packages/circuit-ui/components/Theming/Theming.docs.mdx

```diff
@@ -0,0 +1,168 @@
+  fg: Fg;
+  border: BgOrBorder;
+};
+```
+
+...and as JSON with fake colors, for illustration purposes:
+
+```json
+{
+  "fg": {
+    // neutrals
+    "default": "darkgray",
+    "subtle": "gray",
+    "highlight": "black",
+    // roles (example using primary, same for confirm, notify, alert)
+    "primary": "darkblue",
+    "onPrimary": "white"
+  },
+  "bg": {
+    // neutrals
+    "default": {
+      "idle": "white",
+      "hover": "lightestgray",
+      "active": "lightergray"
+    },
+    "subtle": {
+      "idle": "lightergray",
+      "hover": "lightgray",
+      "active": "gray"
+    },
+    // roles (example using primary, same for confirm, notify, alert)
+    "primary": {
+      "default": {
+        "idle": "darkblue",
+        "hover": "darkerblue",
+        "active": "darkestblue"
+      }
+      // same for other role bg variants, e.g. subtle
+    }
+  },
+  "border": {} // same as bg
+}
+```
+
+### Rationale
+
+- On the top-level, we split foreground colors and background/border colors, because foreground colors have no variants (except neutrals) and states* (*: see open questions)
+- We split neutral colors and role colors, because neutrals typically have more variants (e.g. default neutral text, subtle neutral text, highlight neutral text—but only a single variant of alert text)
+- We split backgrounds and borders because they vary in some cases (e.g. in the light theme, default bg is white and default border is light gray), and also because borders will need to have increased contrast in future high-contrast modes
+
+### Contrast requirements
+
+These can also be programmatically validated in tests.
+
+- Color roles as backgrounds (e.g. `bg.primary.default.*`) need at least 4.5:1 contrast with the accompanying `on*` foreground color (e.g. `fg.onPrimary`)
+- Color roles as foregrounds (e.g. `fg.primary`) need at least 4.5:1 contrast with the default background (`bg.default.*`) (Note: see open question about the notify color role)
+- Subtle variants of color roles (e.g. `bg.primary.subtle.*`) need at least 4.5:1 contrast with neutral fg colors (`fg.default`, `fg.subtle`...)
+- Neutral foregrounds need at least 4.5:1 contrast with neutral backgrounds
+- 🚧
+
+### Open questions
+
+- Can we group text and icon colors under `fg`?
+- Is `promo` another color role, or a different thing?
+- In our current light theme, `fg.notify` (e.g. used in notify icons) doesn't have 4.5:1 contrast to `bg.default`. It would be tricky to change this color—can we rely on documentation _not_ to use notify as text color? Or do we need to split icons from text? (`text.notify` could be set as gray, like in the `Input`'s warning message)
+- Foreground colors currently don't have states in our components, except the `Anchor` (becomes darker when hovered) and the tertiary `Button`. Since the Anchor doesn't need it (an underline is enough) and the tertiary button will become underlined, can we remove states for foregrounds? Or do we need to support them?
```

Comment:
I would assume an underline is enough, since it's enough for a link. It's a very common indicator of interactivity (although users might expect a navigation rather than other interaction).

As far as I know the plan was to switch to the existing "cta button" ([internal Figma ref](https://www.figma.com/file/2yHC0dK9pFEhLsMxk62XZY/Circuit-UI-Web?node-id=606%3A3084)), currently black and underline in light mode. But I see that the specs for it also include hover and active states. I'm not sure what the plan is for icon-only tertiary buttons.

It might make more sense to include states, like you suggested. If there are already some uses of them in our current theme, it's not impossible that they would come up in future themes

Reply:
Added state to foreground colors in [6fc1648](https://github.com/sumup-oss/circuit-ui/pull/1447/commits/6fc164842b6ae4b9ea15179675f53a5ab9f066e5)